experimental/cache_size.cc

```diff
@@ -0,0 +1,65 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <algorithm>
+#include <iostream>
+
+int64_t MAX_SIZE = 16 * 1024 * 1024;
+int64_t MIN_SIZE = 1024;
+int64_t ACCESSES_NUM = 64 * 1024 * 1024;
+int REPEATS = 50;
+
+clock_t whack_cache(const int64_t sz, int64_t* accesses) {
+  char* buf = new char[sz];
+
+  clock_t start = clock();
+  // TODO fencing
+  for (int64_t i = 0; i < ACCESSES_NUM; i++) {
+    ++buf[(accesses[i] * 64) % sz];
+  }
+  // TODO: elapsed time generates zero
```

Comment:
Depending on the optimization level, I can think of a few reasons this might happen:
* the compiler could migrate the call to `clock()` up above the memory accesses
* the compiler may have noticed the loop has no side effects other than changing `buf` but that `buf` is never read again and never leaves this function -- which means it's entitled to throw away the loop entirely.

If it's one of these, you can try using [`ForceRead`](https://github.com/google/safeside/blob/d8a74aec05ecbfdec482049e5beb471ca8e697c6/demos/utils.h#L20), which should prevent the compiler from optimizing them away or moving other function calls ahead of them.

Reply:
Interesting.
I change it to `ForceRead` to see if it resolves.